src/repository/storage/postgres/orm/sequelize/note.ts

```diff
@@ -324,4 +323,20 @@ export default class NoteSequelizeStorage {
       },
     });
   };
+
+  /**
+   * Get all notes based on their ids
+   * @param noteIds - list of note ids
+   */
+  public async getNotesByIds(noteIds: NoteInternalId[]): Promise<Note[]> {
+    const notes: Note[] = await this.model.findAll({
+      where: {
+        id: {
+          [Op.in]: noteIds,
+        },
+      },
+    });
```

Comment:
the order is not guaranteed here. But the order is important. You should add "order by" to ensure that returned notes will have the same order as passed ids

maybe something like this?
```ts
order: [
  Sequelize.literal(`array_position(ARRAY[${noteIds.map(id => `'${id}'`).join(',')}], id)`)
],
```